File: 2_JAVA/Assignment 2/Mathutiltest1.java
class Mathutiltest1{
       public static boolean isEven(int number){
	       if(number%2==0)
	       {
		       return true;
	       }
	       else 
		       return false;
       }


     public static boolean isOdd(int number){
	     if(number%2 !=0)
	     {
		     return true;
	     }
	     else
		     return false;
     }

    public static boolean isPrime(int number){

	    if(number==1)
		    return false;
	    if(number==2 || number==3)
		    return true;
	    if((number%2)==0)
		    return false;
	    for(int i=3; i<=Math.sqrt(number);i+=2)
	    {
		   if((number % i)==0)
			  return false;
	    }
	   return true;
    }

    public static int countPrimes(int first, int last){
	    int term,count=0;

	    for(term=first;term <= last; ++term)
	    {
		    if(isPrime(term) == true)

			    count+=1;
	    }
	    return count;
    }

    public static int reverse(int digit){
	    int rev=0;
	    while (digit!=0){

		    rev =rev*10 + digit % 10;
		    digit/=10;
	    }
	    return rev;
    }

   public static int digitCount(int num){
	   int a=1;
	   int b=num;
	   int c=0;
	   do{
		   a=a*10;
		   c+=1;

	   }while(a<=b);
	   return c;
   }
}

File: Assignment1/Mathutiltest1.cs
    class Mathutiltest1
    {
    public static void Main(string[] args)
    {
        int n =int.Parse(args[0]);
        int num =int.Parse(args[1]);
        int num2 =int.Parse(args[2]);
        int first =int.Parse(args[3]);
        int second=int.Parse(args[4]);
        int digit = int.Parse(args[5]);
        int count = int.Parse(args[6]);
        //Mathutiltest2.IsEven(n);
       //Mathutiltest2.CountPrimes(first,second);

       if(Mathutiltest2.IsEven(n)==true){
		       Console.WriteLine($"the number {n} is even");
	       }
	      else
		       Console.WriteLine($"the number {n} is not even"); 


        if(Mathutiltest2.IsOdd(num)==true){
		       Console.WriteLine($"the number {num} is odd");
	       }
	       else
		     Console.WriteLine($"the number {num} is not odd");   


        if(Mathutiltest2.IsPrime(num2)==true){
		    Console.WriteLine($"the number {num2} is prime");
	    }
	    else
		  Console.WriteLine($"the number {num2} is not prime");


        Console.WriteLine($"the total prime numbers between {first} to {second} are { Mathutiltest2.CountPrimes(first,second)}");

        Console.WriteLine("the digit after reverse is {0}",Mathutiltest2.Reverse(digit));

        Console.WriteLine("the total numbers of digit is {0}",Mathutiltest2.DigitCount(count));
    }


 }
